libraries/rush-lib/src/logic/operations/ShardedPhaseOperationPlugin.ts

```diff
@@ -0,0 +1,212 @@
+
+// eslint-disable-next-line @typescript-eslint/typedef
+const TemplateStrings = {
+  SHARD_INDEX: '{shardIndex}',
+  SHARD_COUNT: '{shardCount}',
+  PHASE_NAME: '{phaseName}'
+} as const;
+
+/**
+ * Phased command that shards a phase into multiple operations.
+ */
+export class ShardedPhasedOperationPlugin implements IPhasedCommandPlugin {
+  public apply(hooks: PhasedCommandHooks): void {
+    hooks.createOperations.tap(PLUGIN_NAME, spliceShards);
+  }
+}
+
+function spliceShards(existingOperations: Set<Operation>, context: ICreateOperationsContext): Set<Operation> {
+  const { projectConfigurations, rushConfiguration } = context;
+
+  const shardableOperations: Set<Operation> = new Set<Operation>();
+
+  for (const operation of existingOperations) {
+    const { associatedPhase: phase, associatedProject: project } = operation;
+    const operationSettings: IOperationSettings | undefined = getOperationSettings(phase, project);
+    if (operationSettings?.sharding) {
+      shardableOperations.add(operation);
+    }
+  }
+  const getCustomParameterValuesForPhase: (phase: IPhase) => ReadonlyArray<string> =
+    getCustomParameterValuesByPhase();
+
+  for (const operation of existingOperations) {
+    const { associatedPhase: phase, associatedProject: project } = operation;
+    const operationSettings: IOperationSettings | undefined = getOperationSettings(phase, project);
+    if (phase && project && operationSettings?.sharding && !operation.runner) {
+      const { count: shards, shardScriptConfiguration } = operationSettings.sharding;
+
+      existingOperations.delete(operation);
+
+      const collatorNode: Operation = new Operation({
+        phase,
+        project
+      });
+      existingOperations.add(collatorNode);
+
+      const parentFolderFormat: string =
+        operationSettings.sharding.outputFolderArgument?.parentFolderNameFormat ??
+        `.rush/operations/${TemplateStrings.PHASE_NAME}/shards`;
+
+      // Create a new one to avoid moving this into `beforeCreateOperation`.
+      const operationMetadataManager: OperationMetadataManager = new OperationMetadataManager({
+        phase,
+        rushProject: project,
+        operation
+      });
+      const parentFolder: string = parentFolderFormat.replace(
+        TemplateStrings.PHASE_NAME,
+        operationMetadataManager.logFilenameIdentifier
+      );
+
+      const collatorDisplayName: string = `${getDisplayName(phase, project)} - collate`;
+      const collatorRawScript: string | undefined = getScriptToRun(project, phase.name, undefined);
+
+      if (collatorRawScript === undefined && phase.missingScriptBehavior === 'error') {
+        throw new Error(
+          `The project '${project.packageName}' does not define a '${phase.name}' command in the 'scripts' section of its package.json`
+        );
+      }
+
+      if (collatorRawScript) {
+        const collatorRunner: ShellOperationRunner = new ShellOperationRunner({
+          commandToRun: collatorRawScript,
```

Comment:
```suggestion
          commandToRun: formatCommand(collatorRawScript, customParameters),
```
Rush custom parameters for the phase should still get forwarded to the collator. They may alter the behavior of the shards in ways that the collator needs to know about.